include/deal.II/matrix_free/matrix_free.templates.h

```diff
@@ -950,7 +950,7 @@ MatrixFree<dim, Number>::initialize_indices(
               if (cell_categorization_enabled)
                 {
                   AssertIndexRange(
-                    cell_it->active_cell_index(),
+                    static_cast<unsigned int>(cell_it->index()),
```

Comment:
The `static_cast` is not necessary here. Otherwise, this looks correct.

Reply:
it is, apparently they have different types.

Reply:
Interesting. Let me fix this in a separate PR.

Reply:
See #7019.

Reply:
This change is correct, apparently a copy-paste issue from the active cell case above...

Reply:
Now, the `static_cast` is not necessary anymore. :wink: